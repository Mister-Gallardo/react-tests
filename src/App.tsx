import { useEffect, useState } from "react"

function App() {
  const [data, setData] = useState('');
  const [toggle, setToggle] = useState(false);
  const [value, setValue] = useState('');

  const onClick = () => setToggle(!toggle);

  useEffect(() => {
    setTimeout(() => {
      setData('hello2');
    }, 100)
  }, []);

  return (
    <div>
      <h1 data-testid='input value'>{value}</h1>
      {toggle && <div data-testid='toggle-elem'>toggle</div>}
      {data && <div style={{color: 'green'}}>{data}</div>}
      <h1>Hello world</h1>
      <button data-testid='toggle-btn' onClick={onClick}>click me</button>
      <input type="text" placeholder="input value" onChange={e => setValue(e.target.value)} />
    </div>
  )
}

export default App
