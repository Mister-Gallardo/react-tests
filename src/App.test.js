import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";

import "@testing-library/jest-dom";
import App from "./App";
import userEvent from "@testing-library/user-event";

describe("TEST APP", () => {
  test("render component", async () => {
    // render(<App />);
    // const helloWorldElem = screen.getByText(/hello world/i);
    // const btn = screen.getByRole("button");
    // const input = screen.getByPlaceholderText(/input value/i);

    // expect(helloWorldElem).toBeInTheDocument();
    // expect(btn).toBeInTheDocument();
    // expect(input).toMatchSnapshot();
    //   screen.debug();
  });

  test("render component", async () => {
    render(<App />);
    //   const helloWorldElem = screen.queryByText(/hello world12345/i);
    //   expect(helloWorldElem).toBeNull();
    // screen.debug();
    // const helloWorldElem = await screen.findByText(/hello2/i);
    // screen.debug();
    // expect(helloWorldElem).toBeInTheDocument();
    // expect(helloWorldElem).toHaveStyle({color: 'green'});
  });

  test("render component", async () => {
    render(<App />);
    
    const btn = screen.getByTestId('toggle-btn');
    expect(screen.queryByTestId('toggle-elem')).toBeNull();
    fireEvent.click(btn);
    expect(screen.queryByTestId('toggle-elem')).toBeInTheDocument();
    fireEvent.click(btn);
    expect(screen.queryByTestId('toggle-elem')).toBeNull();
  });

  test("render component", async () => {
    render(<App />);
    
    const input = screen.getByPlaceholderText(/input value/i);
    expect(screen.getByTestId('input value')).toContainHTML('');
    // fireEvent.input(input, {
    //     target: { value: 'inputt' }
    // })
    await userEvent.type(input, 'inputt');
    expect(screen.getByTestId('input value')).toContainHTML('inputt');
  });
});
