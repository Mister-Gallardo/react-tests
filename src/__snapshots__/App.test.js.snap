// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP render component 1`] = `
<input
  placeholder="input value"
  type="text"
/>
`;

exports[`render component 1`] = `
<input
  placeholder="input value"
  type="text"
/>
`;
